src/sentry/models/exporteddata.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+from datetime import timedelta
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+from uuid import uuid4
+
+from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+
+DEFAULT_EXPIRATION = timedelta(days=7)
+
+
+def generate_data_id():
+    return uuid4().hex
+
+
+class ExportedData(Model):
+    """
+    Stores references to asynchronous data export jobs being stored
+    in the Google Cloud Platform temporary storage solution.
+    """
+
+    __core__ = False
+
+    organization = FlexibleForeignKey("sentry.Organization")
+    user = FlexibleForeignKey(settings.AUTH_USER_MODEL)
+    data_id = models.CharField(max_length=32, unique=True, default=generate_data_id)
+    created_at = models.DateTimeField(default=timezone.now)
+    finished_at = models.DateTimeField(null=True)
+    expired_at = models.DateTimeField(null=True)
+    storage_url = models.URLField(null=True)
+    query_type = models.CharField(max_length=32)  # e.g.: Discover_V1, Billing, IssuesByTag
```

Comment:
@leeandher I don't think a `max_length` field makes sense here.

Reply:
I'm thinking the `query_type` would just be a classifier, with the `query_info` containing more info. I left it as a small charField because the model needs a `max_length` property

Reply:
Does it make more sense for this to be a `smallint` and then represent it with an enum?

Reply:
@wedamija That does make sense, I'm not sure how the representation as an enum part is handled though, how do we connect the two (enums values and integers)

Reply:
You basically have the field be an int, then you have to manually convert it to an Enum, which kind of sucks :(

Reply:
If you had to do this, would use like `models.PositiveSmallInteger`, and an enum with the numbers as the member names or values?

e.g.
```
class ExportQueryType(Enum):
  1= "Discover_V1"
```
or 
```
class ExportQueryType(Enum):
  Discover_V1 = 1
```

and then you'd get the named type with like another `@property` field?

Reply:
It'd make it easier to add choices like this, if we ever wanted to add more and more types without modifying the model, but I've also seen `ChoiceField` implemented in some places, so I'm not sure which to go with